source/WebApp/tests/component.codemirror.mode-asm-instructions.tests.ts

```diff
@@ -0,0 +1,42 @@
+import CodeMirror from 'codemirror';
+import '../components/internal/codemirror/mode-asm';
+
+test('mode-asm-instructions codemirror highlights correctly', async () => {
+
+    document.createRange = () => {
+        const range = new Range();
+
+        range.getBoundingClientRect = jest.fn();
+        range.getClientRects = () => ({
+            item: () => null,
+            length: 0,
+            [Symbol.iterator]: jest.fn()
+        });
+
+        return range;
+    };
+
+    const textarea = document.createElement('textarea');
+    document.body.appendChild(textarea);
+    const cm = CodeMirror.fromTextArea(textarea, {
+        readOnly: true,
+        mode: 'text/x-asm'
+    });
+
+    const assertHighlightedTextFor = (token: string, code: string) => {
+        const renderingListener = (instance: CodeMirror.Editor, line: CodeMirror.LineHandle, element: HTMLElement) =>
+            expect(element.innerHTML).toContain(`<span class="cm-builtin">${token}</span>`);
+
+        cm.on('renderLine', renderingListener);
+        cm.setValue(code);
+        cm.off('renderLine', renderingListener);
+    };
+
+    assertHighlightedTextFor('vmovdqa', `L0011: vmovdqa [rbp-0x30], xmm4
+        L0016: vmovdqa [rbp-0x20], xmm4
+        L001b: vmovdqa [rbp-0x10], xmm4`);
+
+    assertHighlightedTextFor('je', `L000e: je short L0015`);
+
+    assertHighlightedTextFor('jle', `L0015: jle short L0001`); 

```

Comment:
These three to me are separate tests -- you can use `test.each` to run the test with each of those cases, e.g. see `server.get-branches-async.tests`. This might also simplify the assert as you can inline it.

Reply:
I was thinking about using something like xUnit's theory tests in jest, this is it. :)
If we want, we can inject the flat list of all ~2K instructions.